cmd/kubelet/app/server.go

```diff
@@ -1336,17 +1337,18 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	}
 	rl := make(v1.ResourceList)
 	for k, v := range m {
-		switch v1.ResourceName(k) {
-		// CPU, memory, local storage, and PID resources are supported.
-		case v1.ResourceCPU, v1.ResourceMemory, v1.ResourceEphemeralStorage, pidlimit.PIDs:
+		rn := v1.ResourceName(k)
+		switch {
+		// CPU, memory, local storage, PID and hugepages resources are supported.
+		case rn == v1.ResourceCPU, rn == v1.ResourceMemory, rn == v1.ResourceEphemeralStorage, rn == pidlimit.PIDs, v1helper.IsHugePageResourceName(rn):
```

Comment:
why the helper func when the others are strict equality checks?

Reply:
nevermind, the check can't be equality https://github.com/SataQiu/kubernetes/blob/7b1b8d57340e686d013ee7a4180e04b4b16d4606/pkg/api/v1/resource/helpers.go#L301